lib/monitoring/aws-ecs-patterns/BaseServiceMetricFactory.ts

```diff
@@ -1,44 +1,60 @@
-import { BaseService } from "aws-cdk-lib/aws-ecs";
+import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";
+import { IBaseService } from "aws-cdk-lib/aws-ecs";
 
 import { MetricFactory, MetricStatistic } from "../../common";
 
+const EcsNamespace = "AWS/ECS";
 const EcsContainerInsightsNamespace = "ECS/ContainerInsights";
 
 /**
  * Props to create BaseServiceMetricFactory.
  */
 export interface BaseServiceMetricFactoryProps {
-  readonly service: BaseService;
+  readonly service: IBaseService;
 }
 
 /**
  * Metric factory for a base service (parent class for e.g. Fargate and EC2 services).
  */
 export class BaseServiceMetricFactory {
   protected readonly metricFactory: MetricFactory;
-  protected readonly service: BaseService;
+  protected readonly dimensionsMap: DimensionsMap;
+  /**
+   * @deprecated This isn't required by cdk-monitoring-constructs anymore; use your own reference.
+   */
+  protected readonly service: IBaseService;
 
   constructor(
     metricFactory: MetricFactory,
     props: BaseServiceMetricFactoryProps
   ) {
     this.metricFactory = metricFactory;
+    this.dimensionsMap = {
+      ClusterName: props.service.cluster.clusterName,
+      ServiceName: props.service.serviceName,
+    };
     this.service = props.service;
   }
 
   metricClusterCpuUtilisationInPercent() {
-    return this.metricFactory.adaptMetric(
-      this.service.metricCpuUtilization({
-        label: "Cluster CPU Utilization",
-      })
+    return this.metricFactory.createMetric(
+      "CPUUtilization",
+      MetricStatistic.AVERAGE,
```

Comment:
Did the metricFactory used to adapt this metric to `Average` stats before? If not, then would it be a breaking change in regards to the default stats of this metric?

Reply:
It's just a regular `Metric` ([code ref](https://github.com/aws/aws-cdk/blob/166967f11727a28fc11b9af5de0fad6da2a4ad64/packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts#L1222)), which defaults to `Average` ([code ref](https://github.com/aws/aws-cdk/blob/v2.117.0/packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts#L44)).